### PR TITLE
Create node.js.yml

### PR DESCRIPTION
**Pull request by Soreloser2**
Should be ready to go in, although probably not super useful currently as we only have trivial tests (will be more useful with #32)

When adding tests, you shouldn't need to do anything except have the test file in the `/test/` directory (anywhere). If you have a file that is not a test but in the test directory (JS or otherwise), add it to the excluded test files in `run.js`

### CODE REVIEW

**Comment by Soreloser2** [comment]
@paulstn  FYI: May need to make `node.js.yml` a required check once this goes in

File: test/run.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        if (fs.statSync(filepath).isDirectory()) {
+            result.concat(get_files(filepath));
+        } else if (!excluded_test_files.includes(file)) {
+            result.push(filepath);
+        }
+    });
+    return result;
+}
+
+test.run({files: get_files(testDir)}).pipe(process.stdout);
```
Comment (paulstn) [inline test/run.js:27]: Should we pipe an output file instead of stdout? I think either option is fine, just wanted to bring it up as a possibility.

Comment (Soreloser2) [reply test/run.js:27]: We can see all pervious runs on the actions page, and each one's standard output, so it should be saved. We can definitely do this but it should be saved already if we're worried about viewing the details of the runs

Comment (paulstn) [reply test/run.js:27]: Sounds good to me!

**Review by paulstn — APPROVED** [review]

**Review by kashish-ops — APPROVED** [review]
looks good!

**Review by riteshsb — APPROVED** [review]

**Pull request merged**